service/src/main/java/bio/terra/workspace/service/resource/controlled/ControlledResourceMetadataManager.java

```diff
@@ -79,8 +79,10 @@ public ControlledResource validateControlledResourceAndAction(
     ControlledResource controlledResource = resource.castToControlledResource();
     String samName = controlledResource.getCategory().getSamResourceName();
 
+    // Everyone who is a reader (or above) on the workspace can see all the resources in a
+    // workspace. Thus this authorization check is against the workspace and not the resource.
     if (StringUtils.equals(action, SamControlledResourceActions.READ_ACTION)) {
-      checkResourceAuthz(userRequest, samName, resourceId, action);
+      checkResourceAuthz(userRequest, SamConstants.SamResource.WORKSPACE, workspaceUuid, action);
```

Comment:
Should this be checking if the user can read either the workspace OR the resource? Are there cases where a user has read access to a controlled resource, but not the workspace that would start to fail after this change?

Reply:
Since it is a controlled resource, I assume the user has to have access to the workspace to be able to have any access to the resource at all.

FWIW, no tests failed.

Reply:
I think that's the case. There is the notion of a "private resource" that is managed by an application on behalf of a user (think: leo managed azure vms). The private resource user must be still be a reader on the workspace in that case. I'm not familiar with a mechanism by which we can grant access to a single resource on a workspace without first giving a user at least reader access.